prediction_market_agent_tooling/tools/safe.py

```diff
@@ -0,0 +1,126 @@
+from gnosis.safe.safe import Safe
+from loguru import logger
+from safe_cli.safe_addresses import (
+    get_default_fallback_handler_address,
+    get_proxy_factory_address,
+    get_safe_contract_address,
+    get_safe_l2_contract_address,
+)
+from web3.types import Wei
+
+from prediction_market_agent_tooling.tools.hexbytes_custom import HexBytes
+
+
+def create_safe(
+    ethereum_client: EthereumClient,
+    account: LocalAccount,
+    owners: list[str],
+    salt_nonce: int,
+    threshold: int = 1,
+    without_events: bool = False,  # following safe-cli convention
+) -> ChecksumAddress | None:
+    to = NULL_ADDRESS
+    data = b""
+    payment_token = NULL_ADDRESS
+    payment = 0
+    payment_receiver = NULL_ADDRESS
+
+    if len(owners) < threshold:
+        raise ValueError("Threshold cannot be bigger than the number of unique owners")
+
+    ethereum_network = ethereum_client.get_network()
+
+    safe_contract_address = (
+        get_safe_contract_address(ethereum_client)
+        if without_events
+        else get_safe_l2_contract_address(ethereum_client)
+    )
+    proxy_factory_address = get_proxy_factory_address(ethereum_client)
+    fallback_handler = get_default_fallback_handler_address(ethereum_client)
+
+    if not ethereum_client.is_contract(safe_contract_address):
+        raise EnvironmentError(
+            f"Safe contract address {safe_contract_address} "
+            f"does not exist on network {ethereum_network.name}"
+        )
+    elif not ethereum_client.is_contract(proxy_factory_address):
+        raise EnvironmentError(
+            f"Proxy contract address {proxy_factory_address} "
+            f"does not exist on network {ethereum_network.name}"
+        )
+    elif fallback_handler != NULL_ADDRESS and not ethereum_client.is_contract(
+        fallback_handler
+    ):
+        raise EnvironmentError(
+            f"Fallback handler address {fallback_handler} "
+            f"does not exist on network {ethereum_network.name}"
+        )
+
+    account_balance: int = ethereum_client.get_balance(account.address)
+    if not account_balance:
+        logger.info(
+            "Client does not have any funds. Let's try anyway in case it's a network without gas costs"
+        )
+    else:
+        ether_account_balance = round(
+            ethereum_client.w3.from_wei(account_balance, "ether"), 6
+        )
+        logger.info(
+            f"Network {ethereum_client.get_network().name} - Sender {account.address} - "
+            f"Balance: {ether_account_balance} xDAI"
+        )
+
+    if not ethereum_client.w3.eth.get_code(
+        safe_contract_address
+    ) or not ethereum_client.w3.eth.get_code(proxy_factory_address):
+        raise EnvironmentError("Network not supported")
+
+    logger.info(
+        f"Creating new Safe with owners={owners} threshold={threshold} salt-nonce={salt_nonce}"
+    )
+
+    # We ignore mypy below because using the proper class SafeV141 yields an error.
```

Comment:
```suggestion
    # We ignore mypy below because using the proper class SafeV141 yields an error and mypy doesn't understand that there is a hacky `__new__` method on this abstract class.
```

Just for clarity for the future readers

Reply:
Agree, implemented.